### PR TITLE
add config for enum fallback values

### PR DESCRIPTION
**Pull request by knaeckeKami**
as discussed on discord.

Add forward-compatibility of enums via configuration of the build.yaml.
Also, update the README of gql_build.

Since it's not possible to test different build.yaml configs via just one end_to_end_test project, I added tests to gql_builder, I think that makes more sense than adding multiple end_to_end_test projects just for that purpose.

### CODE REVIEW

**Comment by klavs** [comment]
@smkhalsa please review whenever you're available.

File: codegen/gql_code_builder/test/enum/enum_fallback_test.dart [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import "package:test/test.dart";
+
+void main() {
+  final simpleEnum =
+      EnumTypeDefinitionNode(name: NameNode(value: "testEnum"), values: [
+    EnumValueDefinitionNode(name: NameNode(value: "val1")),
+    EnumValueDefinitionNode(name: NameNode(value: "val2"))
+  ]);
+
+  test("does note generate fallback value when no fallback set in config", () {
```
Comment (smkhalsa) [inline codegen/gql_code_builder/test/enum/enum_fallback_test.dart:14]:
```suggestion
  test("does not generate fallback value when no fallback set in config", () {
```

File: codegen/gql_build/README.md [hunk 2]
```diff
@@ -39,14 +39,37 @@ Generates a typed view of your data.
 ### var_builder
 Creates data classes for any GraphQL variables used in a query or fragment.
 
-### enum_builder
-Generates an enum-like class per GraphQL enum type. Defines known enum values to be used in your code, and allows unknown enum values to be used without causing runtime error when handling response data.
+### schema_builder
+Creates data classes from you specified graphql schema.
```
Comment (smkhalsa) [inline codegen/gql_build/README.md:43]:
```suggestion
Creates data classes from your specified graphql schema.
```

**Review by smkhalsa — COMMENTED** [review]
Thanks @ knaeckeKami for the PR! Just a couple of typos, but otherwise looks good.

**Pull request merged**